### ISSUE TITLE
[Bug]: The preview of "Preserve my homepage content" is incorrect

### ISSUE THREAD

**Issue by arthur791004** [Automattic/wp-calypso#69816]
### Quick summary

When you go to "Appearance > Themes", you can select a theme and click the "Activate" button to preview the result. However, the preview of "Preserve my homepage" might be wrong if the current activated theme is a block-based theme.

![image](https://user-images.githubusercontent.com/13596067/200244738-3f03d29a-fc46-49a9-a572-f78b1d30a109.png)

### Steps to reproduce

1. Go to "Appearance > Themes"
2. Activate the theme whose homepage template is `home.html` and don't preserve your homepage content first
    * e.g: Skatepark, Yuga, Archeo, Cultivate, Pendant
4. Go to "Appearance > Themes" again
5. Activate another theme whose homepage template is `home.html` 
     * e.g: Skatepark, Yuga, Archeo, Cultivate, Pendant

### What you expected to happen

The preview for "Preserve my homepage content" is correct

### What actually happened

The preview for "Preserve my homepage content" is wrong

### Related

pbxlJb-2X4-p2#comment-2066

**Comment by cuemarie** [comment]
This report was found in a sweep of backlogged issues that were not prioritized or on the Dotcom Bug Prioritization Board.  Gave it a quick triage test, notes below.

📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Simple – Uncertain

📌 FINDINGS/SCREENSHOTS/VIDEO
- In my test, the preview modal does not reflect my current home template, but this does not quite bely the truth - when I activate a new theme, my home template is switched anyway, even if I chose to preserve my homepage content.
- In other words, The preview for "Preserve my homepage content" is correct, because my home.html template is not preserved in either option.

#### Screenshots/Recordings

https://user-images.githubusercontent.com/27249804/235782994-91338924-b152-4d29-a589-5a207808cee6.mp4

📌 ACTIONS
- Assigned to Lego

📌 Message to Author
- It's possible this issue is no longer relevant, or on hold per pbxlJb-2X4-p2#comment-2338
- With that in mind, not escalating this for now.

**Comment by taipeicoder** [comment]
Address via #77881
